### PR TITLE
Feature/setup dev stack

### LINKED ISSUE #56: Dev stack
**Description**
For local development a stack is needed see https://github.com/it-at-m/refarch-templates/tree/main/stack.
It need to be thought about if a duplication of the refarch-template stack is necessary or an alternative is available.
On copy support for integrations need to be added and not needed services like postgresql should be removed.

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Setup dev stack and add local run configurations.

**Reference**

Issues closes #56

### CODE REVIEW

File: stack/.gitignore
Comment (devtobi) [inline stack/.gitignore:1]: I would keep all gitignore stuff in a single file. Thats why I would move this entry into the `.gitignore` at project root,

File: stack/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    environment:
+      MINIO_ROOT_USER: minio
+      MINIO_ROOT_PASSWORD: Test1234
+    ports:
+      - '9000:9000'
+      - '9001:9001'
+    volumes:
+      - ./minio:/data
+    networks:
+      - internal
```
Comment (devtobi) [inline stack/docker-compose.yml:17]: Could you please add restart policy, security opts and additional health check (if not provided by the image itself) to the additional containers you provided in comparison to the default stack in https://github.com/it-at-m/refarch-templates/blob/main/stack/docker-compose.yml

File: stack/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      "
+    networks:
+      - internal
+
+  # mail
+  mailpit:
+    image: axllent/mailpit:latest
+    ports:
+      - '1025:1025' # smtp server
+      - '8025:8025' # ui
```
Comment (devtobi) [inline stack/docker-compose.yml:37]: Could you please add restart policy, security opts and additional health check (if not provided by the image itself) to the additional containers you provided in comparison to the default stack in https://github.com/it-at-m/refarch-templates/blob/main/stack/docker-compose.yml

File: stack/docker-compose.yml
Comment (devtobi) [inline stack/docker-compose.yml:1]: I would encourage of the Docker compose extends mechanism (see https://docs.docker.com/compose/multiple-compose-files/extends/). This even works for remote files so you could just provide your additional needed containers (like minio and mailpit). This would however mean, that you would also get pgadmin and refarch-gateway inside your stack which might not be what you want.
However it would make things easier in terms of maintaining the stack in the future.

Alternative would be to add a file like `docker-compose-refarch-base.yml` which is a minified version of the refarch-templates stack. Then you provide your additional needs in the `docker-compose.yml` file

File: stack/keycloak/migration/01_init-realm.yml
Comment (devtobi) [inline stack/keycloak/migration/01_init-realm.yml:1]: Again here we have a 1:1 copy from refarch-templates. But don't know how we can avoid this in this case. I guess only solution would be to move everything into 1 repository which might be the case in the future.
However until then it might be fine for now..

File: stack/keycloak/migration/04_client-service-account.yml
Comment (devtobi) [inline stack/keycloak/migration/04_client-service-account.yml:1]: Why is this file needed? This is not part of the `refarch-templates` stack. Might it be suitable for there as well?

File: stack/keycloak/migration/04_client-service-account.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+id: client-service-account
+author: Default
+realm: local_realm
+changes:
+  - updateClient:
+      clientId: local
+      serviceAccountsEnabled: true
+
+# TODO test
```
Comment (devtobi) [inline stack/keycloak/migration/04_client-service-account.yml:9]: Why is a `TODO test` here? Is there something missing?

File: .run/ApiGatewayApplication.run.xml
Comment (devtobi) [inline .run/ApiGatewayApplication.run.xml:1]: What are those `.xml` files used for?

**Review by devtobi — CHANGES_REQUESTED** [review]

File: stack/docker-compose.yml
Comment (simonhir) [reply stack/docker-compose.yml:1]: > This even works for remote files

Where did you see that in my opinion that isn't correct and therefore there is no alternative than copying.

File: stack/keycloak/migration/04_client-service-account.yml (hunk 3 above)
Comment (simonhir) [reply stack/keycloak/migration/04_client-service-account.yml]: Removed it.
Already tested it but forgot to remove it.

File: .run/ApiGatewayApplication.run.xml
Comment (simonhir) [reply .run/ApiGatewayApplication.run.xml:1]: These are run configurations for IntelliJ.
We tried this out with DigiWF and it's very useful as you don't need to configure them yourself and know these should work.

File: stack/keycloak/migration/04_client-service-account.yml
Comment (simonhir) [reply stack/keycloak/migration/04_client-service-account.yml:1]: Service account authentication is needed to authenticate from any other integration against the s3 service.

> Might it be suitable for there as well?
Yes

File: stack/.gitignore
Comment (simonhir) [reply stack/.gitignore:1]: Done

File: stack/docker-compose.yml
Comment (devtobi) [reply stack/docker-compose.yml:1]: Thats a shame. :( But then I'd encourage my suggestions to split into 2 separate parts. One as a minified version of the refarch-templates stack and one with your needed additions. This way its easier to keep in sync.

File: .run/S3IntegrationApplication.run.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="S3IntegrationApplication" type="SpringBootApplicationConfigurationType" factoryName="Spring Boot">
+    <option name="ACTIVE_PROFILES" value="local" />
+    <module name="refarch-s3-integration-rest-service (1)" />
```
Comment (devtobi) [inline .run/S3IntegrationApplication.run.xml:4]: Is this really correct? The `(1)` at the end looks kind of strange to me.

Comment (simonhir) [reply .run/S3IntegrationApplication.run.xml]: Done

File: stack/docker-compose.yml
Comment (simonhir) [reply stack/docker-compose.yml:1]: Done

File: stack/docker-compose.yml (hunk 2 above)
Comment (simonhir) [reply stack/docker-compose.yml:43]: Done

File: stack/docker-compose.yml (hunk 1 above)
Comment (simonhir) [reply stack/docker-compose.yml:17]: Done

File: stack/keycloak/migration/04_client-service-account.yml
Comment (devtobi) [reply stack/keycloak/migration/04_client-service-account.yml:1]: What would be scenario for refarch-templates where this would be useful?

Comment (simonhir) [reply stack/keycloak/migration/04_client-service-account.yml:1]: When they include the cosys-integration-java-starter.
Probably we should just add it when we need it

**Review by devtobi — APPROVED** [review]
LGTM
